enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+// +kubebuilder:validation:XValidation:rule="self.type != 'NLB' || !has(self.classicLoadBalancer)",message="Classic load balancer parameters are allowed only when load balancer type is Classic."
+// +union
+type AWSLoadBalancerParameters struct {
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"nlbParameters,omitempty"`	
+}
+
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+    // You can assign Elastic IP addresses to the Network Load Balancer by adding the following annotation. 
+    // The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+    // service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
+    // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+    // https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations
+    // +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+    EIPAllocations []configv1.EIPAllocations `json:"eipAllocations"`
+}
+
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### 1. Set EIP through installer for the default ingress controller:
+   
+   a. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+   
+   b. The installer then sets the ingress cluster object with the `eipAllocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eipAllocations` from the
+          ingress cluster object.
+     
+     iii. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+          which uses the value from the field `eipAllocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CR:
```

Comment:
This is the proper use case for this text:
```suggestion
#### 2. Create a new ingress controller with EIPs:
```

Reply:
ok